akka-actor/src/main/scala/akka/util/WildcardTree.scala

```diff
@@ -18,15 +18,24 @@ private[akka] final case class WildcardTree[T](data: Option[T] = None, children:
       copy(data = Some(d))
     } else {
       val e = elems.next()
-      copy(children = children.updated(e, children.get(e).getOrElse(WildcardTree()).insert(elems, d)))
+      copy(children = children.updated(e, children.getOrElse(e, WildcardTree()).insert(elems, d)))
     }
 
-  @tailrec final def find(elems: Iterator[String]): WildcardTree[T] =
-    if (!elems.hasNext) this
-    else {
-      (children.get(elems.next()) orElse children.get("*")) match {
-        case Some(branch) ⇒ branch.find(elems)
-        case None         ⇒ WildcardTree()
-      }
+  def find(elems: Iterator[String]): WildcardTree[T] = {
```

Comment:
I'm worried about performance and the amount of objects that are created. This is invoked for each and every actor that is created (and we say that actor creation should be cheap). It wasn't good before either.

Perhaps we should introduce the cache in the `Deployer` now, and if so what kind of cache should be used?